fe/fe-core/src/main/java/com/starrocks/master/ReportHandler.java

```diff
@@ -731,8 +731,15 @@ private static void deleteFromMeta(ListMultimap<Long, Long> tabletDeleteFromMeta
                             continue;
                         }
 
-                        tablet.deleteReplicaByBackendId(backendId);
-                        ++deleteCounter;
+                        // Defer the meta delete to next tablet report, see `Replica.deferReplicaDeleteToNextReport`
+                        // for details.
```

Comment:
if there are more than one old report(network delay), then replica will still be deleted?

Reply:
It could happen in theory, but i think it's quiet a rare case because currently BE reports tablet info with a relatively long interval(every 60s by default) and FE handle tablet report in a single thread and just does some simple computation. For now i haven't thought of a simpler solution without adding some extra syncing mechanism to do this precisely. And if that rare case do happen, it won't cause some serious problem, just some waste of clone work. So i think this pr is simple enough to handle current problem.